Sources/Features/Breadcrumb/BacktraceBreadcrumbs.swift

```diff
@@ -0,0 +1,128 @@
+import Foundation
+
+@objc public enum BacktraceBreadcrumbType: Int {
+
+    case manual = 1
+    case log = 2
+    case navigation = 3
+    case http = 4
+    case system = 5
+    case user = 6
+    case configuration = 7
+
+    var description: String {
+        switch self {
+        case .manual:
+            return "manual"
+        case .log:
+            return "log"
+        case .navigation:
+            return "navigation"
+        case .http:
+            return "http"
+        case .system:
+            return "system"
+        case .user:
+            return "user"
+        case .configuration:
+            return "configuration"
+        }
+    }
+
+    public static let all: [BacktraceBreadcrumbType] = [.manual, .log, .navigation, .http, .system, .user, .configuration]
+
+    public static let none: [BacktraceBreadcrumbType] = []
+}
+
+@objc public enum BacktraceBreadcrumbLevel: Int {
+
+    case debug = 1
+    case info = 2
+    case warning = 3
+    case error = 4
+    case fatal = 5
+
+    var description: String {
+        switch self {
+        case .debug:
+            return "debug"
+        case .info:
+            return "info"
+        case .warning:
+            return "warning"
+        case .error:
+            return "error"
+        case .fatal:
+            return "fatal"
+        }
+    }
+}
+
+@objc public class BacktraceBreadcrumbs: NSObject {
+
+    private var breadcrumbsLogManager: BacktraceBreadcrumbsLogManager?
+    private var backtraceNotificationObserver: BacktraceNotificationObserver?
+    private var breadcrumbLevel: BacktraceBreadcrumbLevel?
+    private var breadcrumbTypes: [BacktraceBreadcrumbType]?
+    private(set) var isBreadcrumbsEnabled: Bool = false
+
+    public func enableBreadcrumbs(_ breadcrumbSettings: BacktraceBreadcrumbSettings = BacktraceBreadcrumbSettings()) {
```

Comment:
shouldn't we return information if we successfully enable breadcrumbs integration? It would be great if our users can somehow prepare themself for situation when the interface is disabled.

Reply:
Although that is a good point, it's the same design as BacktraceMetrics: https://github.com/backtrace-labs/backtrace-cocoa/blob/44e30fa1c9095ddd61090bae9b9aa659185c2b62/Sources/Features/Metrics/BacktraceMetrics.swift#L42

Since the class still can be used (addBreadcrumbs just returns false, but doesn't caused exceptions) I would argue this is okay.

Reply:
Also the user can invoke isBreadcrumbsEnabled property. We can make a ticket to adjust this in both APIs in the future though